File: sdks/python/examples/auth_inventory.py
"""Auth and inventory workflow.

Required env:
    OWLETTE_TOKEN

Optional:
    OWLETTE_API_URL defaults to https://owlette.app
    OWLETTE_SITE_ID overrides the site selection
"""

from __future__ import annotations

import asyncio
import os
import sys

from roost import Roost, RoostApiError


async def main() -> int:
    token = os.environ.get("OWLETTE_TOKEN")
    api_url = os.environ.get("OWLETTE_API_URL") or "https://owlette.app"
    configured_site_id = os.environ.get("OWLETTE_SITE_ID")

    if not token:
        print("missing env var: OWLETTE_TOKEN", file=sys.stderr)
        return 1

    async with Roost(token=token, api_url=api_url) as client:
        try:
            identity = await client.account.whoami()
            version = await client.account.version()
            sites = await client.sites.list()

            print("api", version.current, "supported", ",".join(version.supported))
            print("caller", identity.email or identity.user_id or "api-key")
            print("key", identity.key.key_prefix if identity.key else "session")
            print("sites", len(sites))
            for site in sites:
                print(f"site {site.id} {site.name}")

            site_id = configured_site_id or identity.primary_site_id
            if site_id is None and sites:
                site_id = sites[0].id
            if not site_id:
                print("no site available for machine inventory")
                return 0

            machines = await client.machines.list(site_id)
            print("selected site", site_id, "machines", len(machines))
            for machine in machines:
                heartbeat = machine.last_heartbeat or "never"
                print(
                    f"machine {machine.id} {machine.name} "
                    f"online={machine.online} heartbeat={heartbeat}"
                )
            return 0
        except RoostApiError as err:
            detail = err.problem.get("detail")
            print(f"api error {err.status} {err.code}: {detail}", file=sys.stderr)
            if err.request_id:
                print(f"request_id: {err.request_id}", file=sys.stderr)
            return 1


if __name__ == "__main__":
    raise SystemExit(asyncio.run(main()))

File: sdks/node/examples/auth-inventory.ts
/**
 * Auth and inventory workflow:
 *   1. verify the token with /api/whoami and /api/version
 *   2. list visible sites
 *   3. list machines for the selected site
 *
 * Required env:
 *   OWLETTE_TOKEN or ROOST_TOKEN
 *
 * Optional:
 *   OWLETTE_API_URL or ROOST_BASE defaults to https://owlette.app
 *   OWLETTE_SITE_ID or ROOST_SITE_ID overrides the site selection
 */

import { Owlette, OwletteApiError } from '@owlette/sdk';

const token = process.env.OWLETTE_TOKEN ?? process.env.ROOST_TOKEN;
const apiUrl = process.env.OWLETTE_API_URL ?? process.env.ROOST_BASE ?? 'https://owlette.app';
const configuredSiteId = process.env.OWLETTE_SITE_ID ?? process.env.ROOST_SITE_ID;

if (!token) {
  console.error('missing env var: OWLETTE_TOKEN or ROOST_TOKEN');
  process.exit(1);
}

const owlette = new Owlette({ token, apiUrl });

async function main(): Promise<number> {
  try {
    const [identity, version, sites] = await Promise.all([
      owlette.account.whoami(),
      owlette.account.version(),
      owlette.sites.list(),
    ]);

    console.log('api', version.current, 'supported', version.supported.join(','));
    console.log('caller', identity.email ?? identity.userId ?? 'api-key');
    console.log('key', identity.key?.keyPrefix ?? 'session');
    console.log('sites', sites.length);

    for (const site of sites) {
      console.log(`site ${site.id} ${site.name}`);
    }

    const siteId = configuredSiteId ?? identity.primarySiteId ?? sites[0]?.id;
    if (!siteId) {
      console.log('no site available for machine inventory');
      return 0;
    }

    const machines = await owlette.machines.list(siteId);
    console.log('selected site', siteId, 'machines', machines.length);
    for (const machine of machines) {
      console.log(
        `machine ${machine.id} ${machine.name} online=${machine.online} heartbeat=${machine.lastHeartbeat ?? 'never'}`,
      );
    }
    return 0;
  } catch (err) {
    if (err instanceof OwletteApiError) {
      console.error('api error', err.status, err.code, err.problem.detail ?? err.message);
      if (err.requestId) console.error('request_id', err.requestId);
    } else {
      console.error('unexpected error', err);
    }
    return 1;
  }
}

main().then((code) => process.exit(code));
